### PR TITLE
Server <==> user messages

### PR DESCRIPTION
**Pull request by ianopolous**
This implements a mechanism for a server operator to message users of that server. And for the users to reply, or indeed initiate conversations via feedback. 

This can be used for talking to our users, getting feedback, and surveys. There is a CLI for the server operator to add messages, or read replies from users. 

New messages are shown at login. 

See corresponding web-ui pr: https://github.com/Peergos/web-ui/pull/161

### CODE REVIEW

File: src/peergos/shared/user/ServerConversation.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    public ServerConversation(List<ServerMessage> messages) {
+        this.messages = messages;
+        long latest = 0;
+        for (ServerMessage message : messages) {
+            if (message.sentEpochMillis > latest)
+                latest = message.sentEpochMillis;
+        }
+        ServerMessage lastMessage = messages.get(messages.size() - 1);
+        this.isDismissed = lastMessage.isDismissed || lastMessage.type == ServerMessage.Type.FromUser;
```
Comment (kevodwyer) [inline src/peergos/shared/user/ServerConversation.java:22]: The last part of this statement is surprising (but i can see why you did it).

Comment (ianopolous) [reply src/peergos/shared/user/ServerConversation.java:22]: Isn't it just the logic you asked for?

Comment (kevodwyer) [reply src/peergos/shared/user/ServerConversation.java:22]: Yes. It is the correct logic to filter for display. Using the logic to set a variable called isDismissed is what i question.

Comment (ianopolous) [reply src/peergos/shared/user/ServerConversation.java:22]: Oh sure. I can rename it. isHidden?

Comment (kevodwyer) [reply src/peergos/shared/user/ServerConversation.java:22]: IsDisplayable maybe?

Comment (ianopolous) [reply src/peergos/shared/user/ServerConversation.java]: done

File: src/peergos/server/messages/ServerMessageStore.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.function.*;
+import java.util.logging.*;
+import java.util.stream.*;
+
+public class ServerMessageStore implements ServerMessager {
+    private static final Logger LOG = Logging.LOG();
+
+    private static final String SELECT = "SELECT id, type, sent, body, priorid, dismissed FROM messages WHERE username = ?;";
```
Comment (kevodwyer) [inline src/peergos/server/messages/ServerMessageStore.java:23]: SQL is good at sorting. If returned in chronological order from DB there is less work to do in Java code later.

Comment (ianopolous) [reply src/peergos/server/messages/ServerMessageStore.java:23]: I prefer to keep it simple, and not rely on the db for too much. This results is greater portability across databases.

**Pull request merged**